sierra2mlir/src/statements/mod.rs

```diff
@@ -381,6 +381,16 @@ impl<'ctx> Compiler<'ctx> {
                 .map(|t| (t.ty.get_type(), Location::unknown(&self.context)))
                 .collect_vec(),
         );
+
+        if let Some(gas) = &self.gas {
+            // reduce the gas cost of this user function call
```

Comment:
If I'm inferring intent correctly I think this should read "spend the required gas for the user function call"/"reduce the counter by..." etc, as it's not reducing the cost of the function call itself